README.md

```diff
@@ -2,157 +2,317 @@
+          + [CLI: from source](#cli-from-source-)
+    + [Setup](#setup-)
+* [Usage](#usage-)
+    + [Command-line arguments](#command-line-arguments-)
+    + [File configuration](#file-configuration-)
+    + [TUI](#tui-)
+    + [CLI](#cli-)
+* [For Developers](#for-developers-)
+    + [Build](#build-)
+    + [Test](#test-)
+    + [Run](#run-)
+* [Changelog](#changelog-)
+* [Acknowledgments](#acknowledgments-)
+
+## Demo [↑](#structure)
+
+### TUI demo [↑](#structure)
+
+TODO: TUI demo asciinema
+
+### CLI demo [↑](#structure)
 
 [![asciicast](https://asciinema.org/a/198918.png)](https://asciinema.org/a/198918)
 
-## Getting started
+### Scaffolded project structure [↑](#structure)
 
-### Prerequisites
+This is the example of the project hierarchy you can get if you're running Summoner with all options enabled:
 
-To start using it make sure you have the next tools installed on your machine:
+```
+project-name/
+├── app/
+│   └── Main.hs
+├── benchmark/
+│   └── Main.hs
+├── src/
+│   ├── ProjectName.hs
+│   └── Prelude.hs
+├── test/
+│   └── Spec.hs
+├── CHANGELOG.md
+├── CONTRIBUTING.md
+├── LICENSE
+├── project-name.cabal
+├── README.md
+├── Setup.hs
+├── stack.yaml
+├── appveyor.yml
+├── .git
+├── .gitignore
+├── .stylish-haskell.yaml
+└── .travis.yml
+```
 
-* [`Stack`](http://haskellstack.org) or [`cabal`](https://www.haskell.org/cabal/)
-* [`git`](https://git-scm.com)
-* [`hub`](https://github.com/github/hub)
-* [`curl`](https://curl.haxx.se)
+You can also see complete examples in the following folder:
 
-### Installation
+* [`summon-cli/tests/golden`: Examples of scaffolded projects](https://github.com/kowainik/summoner/tree/master/summoner-cli/test/golden)
 
-The installation process can be done with the one simple command:
+Directory `smallProject` contains an example of the straightforward package, and `fullProject` shows generated project with more advanced structure and more features.
 
-    $ cabal new-install summoner
+## Features [↑](#structure)
 
-or
+Summoner is the tool that combines the predefined configurations, command-line arguments and chosen interface.
 
-    $ stack install summoner
+To fully understand the power in your hands with the help of Summoner, please, check out this section.
 
-or directly from GitHub.
+Below you can see highlighted features in different categories.
 
-You can turn on the bash auto-completion by running the following command:
+### Project [↑](#structure)
+
++ Support for Cabal and Stack build tools.
++ Ability to pick stanzas (library, executable, test-suite, benchmark).
++ Option to include an alternative prelude, if desired. The project then would use [`base-noprelude` technique](http://hackage.haskell.org/package/Prelude), and `Prelude` module would be added to the library target.
++ Whole Hackage-upload checklist support.
++ Support of the multiple GHC versions, with the thoughtful reflection on project meta, base versions (e.g. `base >= 4.9 && < 4.12`), etc.
```

Comment:
Support ~~of the~~ for multiple GHC versions, with ~~the~~ thoughtful reflection on ....